tools/scripts/utils.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  ChildProcess,
+  SpawnOptions,
+  execSync,
+  spawn,
+} from 'node:child_process';
+import { RegistryData, RegistryResult } from './types';
+
+export function configureRegistry({
+  host,
+  registry,
+  registryNoProtocol,
+}: RegistryData) {
+  /**
+   * TODO
+   */
+  // npm
+  process.env.npm_config_registry = registry;
+  // yarnv1
+  process.env.YARN_REGISTRY = registry;
+  // yarnv2
+  process.env.YARN_NPM_REGISTRY_SERVER = registry;
+  console.info(`Set NPM and yarn registry process.env`);
+  /**
+   * TODO
+   */
+  process.env.YARN_UNSAFE_HTTP_WHITELIST = host;
+  console.info(`Set yarn whitelíst process.env`);
+
+  /**
+   * Protocol-Agnostic Configuration: The use of // allows NPM to configure authentication for a registry without tying it to a specific protocol (http: or https:).
+   * This is particularly useful when the registry might be accessible via both HTTP and HTTPS.
+   *
+   * Example: //registry.npmjs.org/:_authToken=your-token
+   */
+  const token = 'secretVerdaccioToken';
+  execSync(`npm config set ${registryNoProtocol}/:_authToken "${token}"`);
+  console.info(`_authToken for ${registry} set to ${token}`);
+}
+
+export function unconfigureRegistry({
+  registryNoProtocol,
+}: Pick<RegistryData, 'registryNoProtocol'>) {
+  execSync(`npm config delete ${registryNoProtocol}/:_authToken`);
+  console.info('delete npm authToken: ' + registryNoProtocol);
+}
+
+export function findLatestVersion(): string {
+  return execSync('git describe --tags --abbrev=0')
+    .toString()
+    .trim()
+    .replace(/^v/, '');
+}
+
+export function parseRegistryData(stdout: string): RegistryData {
+  const port = parseInt(
+    stdout.toString().match(/localhost:(?<port>\d+)/)?.groups?.port,
+  );
```

Comment:
There's a type error here because of potentially assigning `undefined` to a `string` parameter.

![image](https://github.com/user-attachments/assets/b29a2822-2ebe-4cd6-b7d0-01161da7595d)


One solution is to throw for invalid:

```suggestion
  const port = parseInt(
    stdout.toString().match(/localhost:(?<port>\d+)/)?.groups?.port ?? '',
  );
  if (isNaN(port)) {
    throw new Error('Invalid port number');
  }
```

Another option is to define a fallback port:

```suggestion
  const port = parseInt(
    stdout.toString().match(/localhost:(?<port>\d+)/)?.groups?.port ?? '4873',
  );
```